python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +220,52 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate_hashes_signal_type(
+        self,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[t.Type[SignalType]], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> None:
+        if (
+            len(hashes_grouped_by_prefix) > 2
+            and None in hashes_grouped_by_prefix.keys()
+        ):
+            raise CommandError(
+                f"Error: Provided more than one SignalType and some hashes are missing a prefix",
+                2,
+            )
+        if self.only_signal:
+            if (
+                self.only_signal not in hashes_grouped_by_prefix.keys()
+                and None not in hashes_grouped_by_prefix.keys()
+            ):
+                raise CommandError(
+                    f"Error: SignalType '{self.only_signal} was provided, but inferred more from provided hashes."
+                    f"Inferred signal types: {', '.join(s_type.get_name() for s_type in hashes_grouped_by_prefix.keys() if s_type)}"
+                )
+        if (
+            len(signal_types) > 1
+            and len(hashes_grouped_by_prefix) == 1
+            and None in hashes_grouped_by_prefix.keys()
+        ):
+            raise CommandError(
+                f"Error: '{self.content_type.get_name()}' supports more than one SignalType"
+                "No prefix applied to the hashes, cannot infer correct SignalType"
+            )
+        # As well as the above validations, also need to combine the None prefixes into the correct SignalType
+        if None in hashes_grouped_by_prefix.keys():
+            values = set().union(*hashes_grouped_by_prefix.values())
```

Comment:
blocking: Danger! This will merge potentially unrelated hashes together. There is a simpler way to merge this: 

```
if len(signal) != 1:
  throw Exception()
hashes_grouped.setdefault(signals[0], []).extend(hashes_grouped.pop(None))
```